### ISSUE TITLE
Light Client Protocol Dovetailing feature requirements

### ISSUE THREAD

**Issue by nathan-at-least** [zcash/zcash#3675]
From [Announcing Zcash Blossom and proposed feature goals](https://forum.zcashcommunity.com/t/announcing-zcash-blossom-and-proposed-feature-goals/31891):

> # Light Client Protocol Dovetailing
>
> - **What is it?** Alter the base consensus protocol to reinforce light client support.
> - **Who does this affect?** (Future potential) Zcash light client end users.
> - **Why is this a goal?** We anticipate more light client users than full node users in the future (all other things being equal), so any streamlining of this use case in the base consensus protocol is potentially valuable.

**Close Criteria:** Close this ticket when there are well-specified precise requirements for this feature goal.

**Comment by mms710** [comment]
We did a first pass of RICE scoring without the full engineering team (aka we will need to do another pass) and the score for this item was 3.2.

**Comment by mms710** [comment]
Sorry, it took me a while to get back to this. Some context on the RICE scoring method: the RICE score is supposed to help provide a guideline for rough prioritization of various ideas (in this case, various possible Blossom goals). While it will be used as a guideline, it is not a hard and fast rule that something with a higher RICE score will always be worked on first or have a higher priority. The most useful way RICE scores can impact the Zcash selection process is by providing an indication of which items are "low-hanging fruit". You'll notice that some of the higher scored items have low effort associated with them.

In addition, it's important to know that this is an _initial_ RICE score. After the goal is fully detailed and specified, we will perform the RICE scoring again because we will have gained more information that will make the RICE scores more accurate.

We calculate our RICE scores using the following formula:
(Reach * Impact * Confidence)/Effort.
**Reach:** How many users will this change/feature impact, or how many new users will this gain? Alternately, you can think of this as roughly what % of our user base this change will reach. Give this a 1-10 score, 1 being the least amount of reach and 10 being the most. Generally, items with a 9 or 10 score will gain new users.
**Impact:** How large of an impact will this change/feature have on users? Will it increase their usage of Zcash, or of shielded transactions? Give this a 1-10 score, 1 being the least impact and 10 being the most.
**Confidence**: How confident are you that this change will reach the amount of users you said and impact users as much as you said? Please express this in a % form between 1-100% (typically in increments of 10), 10% meaning not at all confident and 100% being very confident.
**Effort**: How many person months will this work take to complete? This should be an estimate and you can use portions of person months in the calculation. For example, 2 weeks = .5 months.

**RICE Score for this Item:**
Reach: 9 - this will impact a good portion of our users and may gain us new users
Impact: 5 - we expect this to have a moderate impact; many users will likely see performance improvements or similar nice-to-haves
Confidence: 70% - we're somewhat confident about our reach and impact numbers
Effort: 10 - we think this will take 10 person months to complete

Calculation: (9* 5 *.7)/10 = **3.2 RICE Score**

**Comment by daira** [comment]
It's unclear to me which improvements are included in this ticket (we haven't done sufficient design work yet, and what we have done hasn't been written down), so I have low confidence in these Reach, Impact and Effort estimates.

**Comment by mms710** [comment]
We need to get further along in our wallet project before we can properly flesh out requirements for this. Moving to the NU3 upgrade for now.

**Comment by nathan-at-least** [comment]
I think this ticket is too general, and I will close it. Here is a set of more specific tickets which could be argued to fall under this umbrella:

- https://github.com/zcash/zcash/issues/288
- https://github.com/zcash/zcash/issues/3835
- https://github.com/zcash/zcash/issues/3812
- https://github.com/zcash/zcash/issues/3638
- https://github.com/zcash/zcash/issues/3226
